### PR TITLE
Chore/submit result

### PR DESCRIPTION
**Pull request by allanbrondumkr**
Contents:
* `TransactionResult` enum and `SubmitResponse`
* support for deserializing `Memo` in `xrpl_api::TransactionCommon`
* added `xrpl_types::Currency` to represent issued tokens and XRP
* added `to_dex`/`from_hex` methods and `Debug` hex implementation to `Hash` types and `Blob` type and `Debug` hex implementation for `AccountId`
* fixed some clippy warnings

### CODE REVIEW

File: xrpl_api/src/types/transaction/common.rs [hunk 1]
```diff
@@ -1,49 +1,211 @@
...
     pub fee: String,
     pub sequence: u32,
-    #[serde(rename = "AccountTxnID")]
+    #[serde(rename = "AccountTxnID", skip_serializing_if = "Option::is_none")]
     pub account_txn_id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub last_ledger_sequence: Option<u32>,
-    // pub memos: Option<Vec<Memo>>,
-    pub memos: Option<Vec<serde_json::Value>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (allanbrondumkr) [inline xrpl_api/src/types/transaction/common.rs:15]: added this in many places where it was missing

File: xrpl_binary_codec/src/error.rs [hunk 2]
```diff
@@ -1,34 +1,11 @@
-use std::num::ParseIntError;
 use thiserror::Error;
 
 #[derive(Error, Debug)]
 pub enum BinaryCodecError {
-    #[error("Error when parsing field: {0}")]
```
Comment (allanbrondumkr) [inline xrpl_binary_codec/src/error.rs:6]: These error variants were unused, so removed them

File: xrpl_api/src/types/transaction/common.rs [hunk 3]
```diff
@@ -1,49 +1,211 @@
...
+            memo_data: Option<&'a str>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            memo_format: Option<&'a str>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            memo_type: Option<&'a str>,
+        }
+
+        #[derive(Serialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct MemoElmRaw<'a> {
```
Comment (gmosx) [inline xrpl_api/src/types/transaction/common.rs:109]: What does 'Elm' stand for?

Comment (allanbrondumkr) [reply xrpl_api/src/types/transaction/common.rs:109]: Just `Element`, should have typed it out fully. It it they outer layer in the weird array representation with the Element(Object(..)) structure

**Pull request merged**